databricks/koalas/config.py

```diff
@@ -28,33 +28,87 @@
 __all__ = ['get_option', 'set_option', 'reset_option']
 
 
-# dict to store registered options and their default values (key -> default).
-_registered_options = {
-    # This sets the maximum number of rows koalas should output when printing out various output.
-    # For example, this value determines whether the repr() for a dataframe prints out fully or
-    # just a truncated repr.
-    "display.max_rows": 1000,  # TODO: None should support unlimited.
-
-    # `plotting.max_rows` sets the visual limit on TopN plots. If it is set to 1000, the first 1000
-    # data points will be used for plotting, and this number will be seen on the right up corner.
-    "plotting.max_rows": 1000,
-
-    # 'compute.max_rows sets the limit of the current DataFrame. Set `None` to unlimit
-    # the input length. When the limit is set, it is executed by the shortcut by collecting
-    # the data into driver side, and then using pandas API. If the limit is unset,
-    # the operation is executed by PySpark. Default is 1000.
-    "compute.max_rows": 1000,  # TODO: None should support unlimited.
-
-    # This determines whether or not to operate between two different dataframs.
-    # For example, 'combine_frames' function internally performs a join operation which can be
-    # expensive in general.
-    # So, if `compute.ops_on_diff_frames` variable is not True, that method throws an exception.
-    "compute.ops_on_diff_frames": False,
-
-    # This sets the default index type: sequence, distributed and distributed-sequence.
-    "compute.default_index_type": "sequence",
-}  # type: Dict[str, Any]
-
+class Option:
+    def __init__(
+            self,
+            *,
+            key: str,
+            doc: str,
+            default: Any,
+            types: Union[Tuple[type, ...], type] = str,
+            check_func: Tuple[Callable[[Any], bool], str] = (lambda v: True, "")):
+        self.key = key
+        self.doc = doc
+        self.default = default
+        self.types = types
+        self.check_func = check_func
+
+    def validate(self, v: Any) -> None:
+        if not isinstance(v, self.types):
+            raise ValueError("The value for option '%s' was %s; however, expected types are "
+                             "[%s]." % (self.key, type(v), str(self.types)))
+        if not self.check_func[0](v):
+            raise ValueError(self.check_func[1])
+
+
+# Available options.
+_options = [
+    # TODO: None should support unlimited.
+    Option(
+        key='display.max_rows',
+        doc=(
+            "This sets the maximum number of rows koalas should output when printing out "
+            "various output. For example, this value determines whether the repr() for a "
+            "dataframe prints out fully or just a truncated repr."),
+        default=1000,
+        types=int,
+        check_func=(lambda v: v >= 0, "'display.max_rows' should be larger than 0.")),
```

Comment:
maybe `lambda v: v>0`? since in the error message it says `should be larger than 0`.
Or alternatively, rephrase the error message a bit?

the same below